### PR TITLE
feat(build-tools): Add generate:node10Entrypoints command

### PR DESCRIPTION
**Pull request by sonalideshpandemsft**
This PR add a new command to generate node10 type declaration entrypoints. Most of the node10 code is extracted from [generate entrypoint](https://github.com/microsoft/FluidFramework/blob/main/build-tools/packages/build-cli/src/library/commands/generateEntrypoints.ts) command.

### CODE REVIEW

File: build-tools/packages/build-cli/src/library/index.ts
Comment (tylerbutler) [inline build-tools/packages/build-cli/src/library/index.ts:63]: As in the other PR, just import directly from the submodules and remove the export here.

File: build-tools/packages/build-cli/src/commands/generate/node10Entrypoints.ts
Comment (tylerbutler) [inline build-tools/packages/build-cli/src/commands/generate/node10Entrypoints.ts:114]: Since you need these util functions in both commands now, put them somewhere in library.

File: build-tools/packages/build-cli/src/commands/generate/node10Entrypoints.ts
Comment (tylerbutler) [inline build-tools/packages/build-cli/src/commands/generate/node10Entrypoints.ts:128]: Is this the same as the exported function from library/commands/? Any reason not to use the exported function?

File: build-tools/packages/build-cli/src/commands/generate/node10Entrypoints.ts
Comment (tylerbutler) [inline build-tools/packages/build-cli/src/commands/generate/node10Entrypoints.ts:51]: Should be replaceable with a shared function, maybe even the one in generate:typetests.

File: build-tools/packages/build-cli/src/commands/generate/node10Entrypoints.ts
Comment (sonalideshpandemsft) [reply build-tools/packages/build-cli/src/commands/generate/node10Entrypoints.ts:128]: My understanding is `generateEntrypoints` will be deprecated and eventually removed once `sourceEntrypoints` is merged. It doesn't seem appropriate for the Node 10 function to be included under `generateSourceEntrypoints`, and I believe it would be better to isolate Node 10 functions since there are not many of them.

File: build-tools/packages/build-cli/src/commands/generate/node10Entrypoints.ts
Comment (tylerbutler) [inline build-tools/packages/build-cli/src/commands/generate/node10Entrypoints.ts:60]:
```suggestion
	if (packageJson.exports === undefined) {
		throw new Error('no valid "exports" within package properties');
	}
```

File: build-tools/packages/build-cli/src/commands/generate/node10Entrypoints.ts
Comment (tylerbutler) [inline build-tools/packages/build-cli/src/commands/generate/node10Entrypoints.ts:65]: Why is this needed?

File: build-tools/packages/build-cli/src/commands/generate/node10Entrypoints.ts
Comment (tylerbutler) [inline build-tools/packages/build-cli/src/commands/generate/node10Entrypoints.ts:9]: Use the PackageJson type from @fluidframework/build-tools

File: build-tools/packages/build-cli/src/commands/generate/node10Entrypoints.ts
Comment (tylerbutler) [inline build-tools/packages/build-cli/src/commands/generate/node10Entrypoints.ts:19]: nit: not a blocker. Ideally either the path to the folder or package.json would be a flag, even if it has a default
value. Makes it easier to test. But it can always be added later.

File: build-tools/packages/build-cli/src/commands/generate/node10Entrypoints.ts
Comment (sonalideshpandemsft) [reply build-tools/packages/build-cli/src/commands/generate/node10Entrypoints.ts]: removed this condition with the change you recommended

**Review by tylerbutler — APPROVED** [review]
Approved. I suggest adding these two input flags from the existing command: --outDir and --mainEntrypoint. Those will give us more flexibility when we start using this. But no need to block on this if you do a follow-up. We could still get this merged as-is.

**Pull request merged**
